dask/dataframe/io/parquet/arrow.py

```diff
@@ -1168,23 +1132,34 @@ def _construct_collection_plan(cls, dataset_info):
                 filter_columns.add(col)
 
         # Determine which columns need statistics.
+        # At this point, gather_statistics is only True if
+        # the user specified calculate_divisions=True
         stat_col_indices = {}
+        _index_cols = index_cols if (gather_statistics and len(index_cols) == 1) else []
         for i, name in enumerate(schema.names):
-            if name in index_cols or name in filter_columns:
+            if name in _index_cols or name in filter_columns:
                 if name in partition_names:
                     # Partition columns wont have statistics
                     continue
                 stat_col_indices[name] = i
 
-        # If the user has not specified `gather_statistics`,
-        # we will only do so if there are specific columns in
-        # need of statistics.
-        # NOTE: We cannot change `gather_statistics` from True
-        # to False (even if `stat_col_indices` is empty), in
-        # case a `chunksize` was specified, and the row-group
-        # statistics are needed for part aggregation.
-        if gather_statistics is None:
-            gather_statistics = bool(stat_col_indices)
+        # We now have enough info to set the final gather_statistics
+        gather_statistics = None if not gather_statistics else True
+        if (
+            chunksize
+            or (int(split_row_groups) > 1 and aggregation_depth)
+            or filter_columns.intersection(stat_col_indices)
+        ):
+            # Need to gather statistics if we are aggregating files
+            # or filtering
+            # NOTE: Should avoid gathering statistics when the agg
+            # does not depend on a row-group statistic
+            gather_statistics = True
+        elif not stat_col_indices or gather_statistics is None:
+            # Not aggregating files/row-groups.
+            # We only need to gather statistics if `stat_col_indices`
+            # is populated
+            gather_statistics = False
```

Comment:
What does `gather_statistics = None` indicate below here? Since this is all internal, I would have thought this should be a bool alone after this point.

Reply:
I was able to clean up this logic a bit and move it to a shared `_set_gather_statistics` function in `parquet/utils.py`.  We want to make sure `gather_statistics` is either `True` or `False` once this block is finished. When we enter this block, we only want `gather_statistics=True` if the user specified `calculate_divisions=True` at the top level. Otherwise, we want to start with `False` (I removed the confusing `None` logic).